documentation/src/main/antora/modules/ROOT/pages/tutorial-create-an-output-component.adoc

```diff
@@ -0,0 +1,212 @@
+                       final HazelcastService service) {
+    this.configuration = configuration;
+    this.jsonFactory = jsonFactory;
+    this.jsonb = jsonb;
+    this.service = service;
+}
+----
+
+=== The PostConstruct method
+[source,java,indent=0,subs="verbatim,quotes,attributes"]
+----
+private transient HazelcastInstance instance;
+private transient IMap<Object, Object> map;
+
+@PostConstruct
+public void init() {
+    instance = service.findInstance(configuration.newConfig()); <1>
+    map = instance.getMap(configuration.getMapName()); <2>
+}
+----
+We will need Hazelcast instance and Hazelcast map. We add those as attributes to the output.
+<1> Here we create an instance of hazelcast according to the provided configuration.
+Here you can notice that we use the injected HazelcastService instance to perform that.
+This service is implemented in the project. See the implementation in xref:tutorial-create-an-input-component.adoc#_the_postconstruct_method["Create an input component" tutorial].
+<2> We get the Hazelcast map according to map name from configuration. We use Hazelcast instance for that purpose.
+
+NOTE: in production you will not want to create one instance per thread/worker but we will cover that in another coming tutorial
+
+=== The PreDestroy method
+[source,java,indent=0,subs="verbatim,quotes,attributes"]
+----
+@PreDestroy
+public void close() {
+    instance.getLifecycleService().shutdown();
+    map = null;
+}
+----
+We shutdown the instance that we have created in the PostConstruct and we free the Hazelcast map reference.
+
+=== The ElementListener method
+[source,java,indent=0,subs="verbatim,quotes,attributes"]
+----
+@ElementListener
+public void onElement(final JsonObject defaultInput) { <1>
+    final Object key = toValue(defaultInput.get(configuration.getKeyAttribute()));
+    final Object value = toValue(defaultInput.get(configuration.getValueAttribute()));
+    map.put(key, value);
+}
+
+private Object toValue(final JsonValue jsonValue) { <2>
+    if (jsonValue == null) {
+        return null;
+    }
+    if (jsonValue.getValueType() == STRING) {
+        return JsonString.class.cast(jsonValue).getString();
+    }
+    if (jsonValue.getValueType() == NUMBER) {
+        return JsonNumber.class.cast(jsonValue).doubleValue();
+    }
+    return jsonValue.asJsonObject();
+}
+----
+<1> This method will be used to pass the incoming data to our output.
+Every object passed should be a JsonObject instance.
+This method can include any logic required to write data to the data source.
+In our implementation we will put data to Hazelcast map.
+
+<2> This is our inner method which is used to transform incoming values in format
+required to put data to Hazelcast map.
+
+=== The full implementation of the Output
+Here is the full code source for the output to have a global view of it.
+xref:component-definition.adoc#_output[Read more about output...]
+[source,java,indent=0,subs="verbatim,quotes,attributes"]
```

Comment:
Here, can you please add `role="initial-block-closed"` after attributes. so that the hole impl is shown collapsed.